src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts

```diff
@@ -99,6 +99,7 @@ export class XtermTerminal extends DisposableStore implements IXtermTerminal {
 		rows: number,
 		location: TerminalLocation,
 		private readonly _capabilities: ITerminalCapabilityStore,
+		disableShellIntegrationTelemetry: boolean | undefined,
```

Comment:
This and the terminalInstance/ptyService changes can be removed as well I believe.

Reply:
we only want to do this for the renderer side under certain conditions, so that's still needed

Reply:
so we'll only do this when there's a telemetry service (renderer) and it's a feature terminal, has no executable or is hide from user